ie-solver/ie_solver_tools.h

```diff
@@ -0,0 +1,84 @@
+#ifndef _IE_SOLVER_TOOLS_H_
```

Comment:
To see proper diff, change name back temporarily, then perform name change in subsequent commit